Source/BestProjectEver/Private/Character/BPE_PlayerCharacter.cpp

```diff
@@ -585,6 +588,85 @@ void ABPE_PlayerCharacter::PlayUltimateEffects()
 	}
 }
 
+//----------------------------------------------------------------------------------------------------------------------
+void ABPE_PlayerCharacter::Reload()
+{
+	if(AreGameplayInputsEnabled() && CarriedAmmo > 0)
+	{
+		if(HasAuthority())
+		{
+			HandleReloading();
+		}
+		else
+		{
+			Server_Reload();
+		}
+	}
+}
+
+//----------------------------------------------------------------------------------------------------------------------
+void ABPE_PlayerCharacter::Server_Reload_Implementation()
+{
+	Reload();
+}
+
+//----------------------------------------------------------------------------------------------------------------------
+bool ABPE_PlayerCharacter::Server_Reload_Validate()
+{
+	return true;
+}
+
+//----------------------------------------------------------------------------------------------------------------------
+void ABPE_PlayerCharacter::HandleReloading()
+{
+	if(IsValid(CurrentWeapon) && !CurrentWeapon->IsMagFull())
+	{
+		const int32 AvailableSpaceInMag = CurrentWeapon->GetMagCapacity() - CurrentWeapon->GetCurrentAmmo();
+		const int32 AmmoToReload = FMath::Min(AvailableSpaceInMag, CarriedAmmo);
+		
+		CarriedAmmo = FMath::Clamp(CarriedAmmo - AmmoToReload, 0, MaxCarriedAmmo);
+		OnCarriedAmmoUpdate();
+
+		CurrentWeapon->OnReloading(AmmoToReload);
+	}
+}
+
+//----------------------------------------------------------------------------------------------------------------------
+void ABPE_PlayerCharacter::TakeWeaponAmmo(ABPE_Weapon* Weapon)
+{
+	if(IsValid(Weapon) && !Weapon->IsEmpty())
+	{
+		const int32 AvailableSpaceCarriedAmmo = MaxCarriedAmmo - CarriedAmmo;
+		const int32 AmmoToReload = FMath::Min(AvailableSpaceCarriedAmmo, Weapon->GetCurrentAmmo());
+		
+		CarriedAmmo = FMath::Clamp(CarriedAmmo + AmmoToReload, 0, MaxCarriedAmmo);
+		OnCarriedAmmoUpdate();
+		
+		Weapon->OnReloading(-AmmoToReload);
```

Comment:
Reloading is reloading. You don't reload negative numbers. If you want to remove bullets from a weapon, you should not handle that on your reloading function.

Reply:
![image](https://user-images.githubusercontent.com/85327912/176237426-f14ed80a-8d54-4814-915c-b2765d8d5c10.png)

Reply:
![image](https://user-images.githubusercontent.com/85327912/176237545-2a35d6b2-9ae2-43cc-80d9-55e9637f7b98.png)

Reply:
![image](https://user-images.githubusercontent.com/85327912/176238916-bf034f7d-77ca-4fa9-a792-941f27f059b8.png)

Reply:
![image](https://user-images.githubusercontent.com/85327912/176239806-18c1e63a-1176-40fe-9d38-336cc94c8d22.png)